periodictable/mass.py

```diff
@@ -127,2943 +127,2943 @@ def init(table, reload=False):
-102-No-257,257.096850(30),,[259]
-102-No-258,258.09820(22)#,,[259]
-102-No-259,259.10102(11)#,,[259]
-102-No-260,260.10264(22)#,,[259]
-102-No-261,261.10574(32)#,,[259]
-102-No-262,262.10752(58)#,,[259]
-103-Lr-251,251.09436(32)#,,[262]
-103-Lr-252,252.09533(32)#,,[262]
-103-Lr-253,253.09526(24)#,,[262]
-103-Lr-254,254.09659(36)#,,[262]
-103-Lr-255,255.09677(22)#,,[262]
-103-Lr-256,256.09876(24)#,,[262]
-103-Lr-257,257.09961(22)#,,[262]
-103-Lr-258,258.10188(11)#,,[262]
-103-Lr-259,259.102990(80)#,,[262]
-103-Lr-260,260.10557(12)#,,[262]
-103-Lr-261,261.10694(22)#,,[262]
-103-Lr-262,262.10969(32)#,,[262]
-103-Lr-263,263.11139(39)#,,[262]
-104-Rf-253,253.10068(49)#,,[261]
-104-Rf-254,254.10017(31)#,,[261]
-104-Rf-255,255.10149(22)#,,[261]
-104-Rf-256,256.101180(29),,[261]
-104-Rf-257,257.10307(29)#,,[261]
-104-Rf-258,258.10357(22)#,,[261]
-104-Rf-259,259.105630(80)#,,[261]
-104-Rf-260,260.10643(22)#,,[261]
-104-Rf-261,261.10875(11)#,,[261]
-104-Rf-262,262.10992(30)#,,[261]
-104-Rf-263,263.11254(20)#,,[261]
-104-Rf-264,264.11398(48)#,,[261]
-105-Db-255,255.10740(45)#,,[262]
-105-Db-256,256.10811(39)#,,[262]
-105-Db-257,257.10786(25)#,,[262]
-105-Db-258,258.10944(37)#,,[262]
-105-Db-259,259.10972(31)#,,[262]
-105-Db-260,260.11143(25)#,,[262]
-105-Db-261,261.11211(25)#,,[262]
-105-Db-262,262.11415(20)#,,[262]
-105-Db-263,263.11508(18)#,,[262]
-105-Db-264,264.11747(25)#,,[262]
-105-Db-265,265.11866(30)#,,[262]
-106-Sg-258,258.11315(45)#,,[266]
-106-Sg-259,259.11465(23)#,,[266]
-106-Sg-260,260.114440(40),,[266]
-106-Sg-261,261.11620(30)#,,[266]
-106-Sg-262,262.11648(30)#,,[266]
-106-Sg-263,263.11831(13)#,,[266]
-106-Sg-264,264.11892(30)#,,[266]
-106-Sg-265,265.12107(15)#,,[266]
-106-Sg-266,266.12193(31)#,,[266]
-107-Bh-260,260.12180(66)#,,[264]
-107-Bh-261,261.12180(26)#,,[264]
-107-Bh-262,262.12301(40)#,,[264]
-107-Bh-263,263.12315(45)#,,[264]
-107-Bh-264,264.12473(30)#,,[264]
-107-Bh-265,265.12520(41)#,,[264]
-107-Bh-266,266.12701(38)#,,[264]
-107-Bh-267,267.12774(37)#,,[264]
-108-Hs-263,263.12871(39)#,,[277]
-108-Hs-264,264.128410(50),,[277]
-108-Hs-265,265.13000(32)#,,[277]
-108-Hs-266,266.13004(44)#,,[277]
-108-Hs-267,267.13177(11)#,,[277]
-108-Hs-268,268.13216(44)#,,[277]
-108-Hs-269,269.13411(46)#,,[277]
-108-Hs-277,,,[277]
-109-Mt-265,265.13657(50)#,,[268]
-109-Mt-266,266.13794(38)#,,[268]
-109-Mt-267,267.13753(62)#,,[268]
-109-Mt-268,268.13882(34)#,,[268]
-109-Mt-269,269.13911(59)#,,[268]
-109-Mt-270,270.14072(66)#,,[268]
-109-Mt-271,271.14123(65)#,,[268]
-110-Ds-267,267.14396(41)#,,[281]
-110-Ds-268,268.14353(54)#,,[281]
-110-Ds-269,269.14514(31)#,,[281]
-110-Ds-270,270.14463(70)#,,[281]
-110-Ds-271,271.14608(20)#,,[281]
-110-Ds-272,272.14631(70)#,,[281]
-110-Ds-273,273.14925(47)#,,[281]
-110-Ds-281,,,[281]
-111-Rg-272,272.15348(36)#,,[272]
-112-Cn-285,,,[285]
-113-Nh-286,,,[286]
-114-Fl-289,,,[289]
-115-Mc-289,,,[289]
-116-Lv-293,,,[293]
-117-Ts-294,,,[294]
-118-Og-294,,,[294]\
+1-H-1,1.0078250319000(100),99.9885(70),1.00794(7)
```

Comment:
(1) The trailing zeros on the value and uncertainty are suspicious. I doubt the uncertainty is known to three digits of accuracy, and I doubt the best estimate of the value with the given uncertainty happens to end in "00".

I understand why they have trailing digits in the rounded table for 1-H-4 for example. They are scaling by 1e6, so they need the extra zeros to reach the decimal point. Where they have uncertainty after the decimal they are only using as many uncertainty digits as they need.

The rule seem to be:

    two digits for uncertainties between …00095… and …00294…, represented as (10), (11), … (29)
    one digit for uncertainties between …00296… and …00949…, represented as (3), (4), …, (9)

For your table you would need to strip zeros from the value and the uncertainty until the uncertainty is between 3 and 29. [Reviewing the table it seems they are keeping two digits for the uncertainties on their non-experimental values.]

However, we are not doing anything with the uncertainties and even if we were, the floating point representation for 0.003000 and 0.003 are indistinguishable, so this is an aesthetic rather than a functional request. If you choose to leave them in drop some text in front of the table indicating that you know they are not meaningful.

(2) You have left the '#' in for the non-experimental values. I'm guessing this doesn't trip up the parser because it is after the uncertainty? Add a comment that says the hash in the field indicates computed rather than measured values.

(3) Did you check if any values changed enough modify the mass and uncertainty of the isotopic average?

Some spot checking suggests it is unlikely.

A full check sounds like work:
```
current element is undefined
for each line:
    parse new element, mass+uncertainty, abundance+uncertainty
    if current element not equal new element
        if current element is defined
            check if accumulated mass matches natural abundance mass for the current element
        set current element to new element
        set accumulated mass to zero
        parse natural abundance mass+uncertainty for the new element
    if abundance > 0, accumulate mass times abundance with uncertainty
```

Reply:
(1) I tried a few things, for the trailing zeroes, but it is not trivial since there are a number of instances where the uncertainty is quoted in the paper with more than two significant figures (most often in the cases of partially calculated values), or e.g. H-4 which has a value of 4.026430+/-110. I have added a note stating that the trailing zeroes do not describe the known uncertainty.

(2) There were already '#'s in the table so i decided it was best to continue the schema. I have added a note to the docstring detailing that these values are not solely experimentally derived.

(3) For the isotopic averages, I noticed an interesting feature, that the uncertainty of the natural abundance mass does not match the uncertainties of the natural abundances. Take for example Li, natural abundance mass is 6.941(2) - an uncertainty of 0.029%, however the abundances are Li-6: 7.59(4), and Li-7: 92.41(4) (total abundance is therefore 100.0(5.7) - e.g. an uncertainty of .57%). Therefore, if i propagate the quoted uncertainties in the abundances, there are quite a few instances where the values disagree (Li being the first). I can ignore the uncertainty propagation from the abundances, or include them and will end up changing quite a few of the values, with the effect of probably increasing the uncertainty of many of the values. Your thoughts on this are most welcome.

Reply:
(1) The ±110 value for H-4 is expected: 107.354 rounds to 110, giving two digits for uncertainties starting with 1 or 2.

Agreed that the uncertainties on the calculated values look a bit strange.

(2) The '#' in the table are fine.

I would have put the comment above the table rather than as part of the public docstring since we are ignoring the uncertainties, but no problem with leaving it where it is. If nothing else it reminds the user that the uncertainties could be made available if needed.

There is a small typo (`vlaues` rather than `values`). I'll fix it after merging.

(3) Thanks for looking into this.

I think what we have now is okay. Consider the case of two stable isotopes. Take n samples, find the relative quantity of `a` and `b` in each. By definition `a+b = 1` since abundance adds to 100%. Because `a` and `b` are correlated it is not surprising that the uncertainty in natural mass from `a*ma + b*mb` does not match the propagated uncertainty from `a` and `b` treated as if they are independently distributed.

A quick numerical simulation with extreme variability `a ~ U[0.0, 0.1]` and isotope masses of 4 and 5 gave an natural mass of 4.950(28) but a value of 4.95(18) with uncertainty propagation. The mean mass was identical to full precision in the two cases.